src/layout/layout.js

```diff
@@ -303,7 +303,7 @@
 
           var drawerButtonIcon = document.createElement('i');
           drawerButtonIcon.classList.add(this.CssClasses_.ICON);
-          drawerButtonIcon.textContent = this.Constant_.MENU_ICON;
+          drawerButtonIcon.innerHTML = this.Constant_.MENU_ICON;
```

Comment:
Actually. I'm not sure it matters either way. If people can modify the
prototype at all then they can inject a script. Or modify the prototype so
it doesn't filter. The only saving grace here is developers using CSP
properly and in enforcement mode.

So ignore this thinking, way to early morning.
On Jan 6, 2016 9:55 AM, "Surma" <notifications@github.com> wrote:

> In src/layout/layout.js
> <https://github.com/google/material-design-lite/pull/1958#discussion_r48964685>
> :
>
> > @@ -303,7 +303,7 @@
> >
> >            var drawerButtonIcon = document.createElement('i');
> >            drawerButtonIcon.classList.add(this.CssClasses_.ICON);
> > -          drawerButtonIcon.textContent = this.Constant_.MENU_ICON;
> > +          drawerButtonIcon.innerHTML = this.Constant_.MENU_ICON;
>
> I’m security n00b. Could you explain to me how anything can get evaluated
> if I block replace < and > with &lt; and &gt; respectively?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/material-design-lite/pull/1958/files#r48964685>
> .
>